Tools/CISupport/ews-build/steps.py

```diff
@@ -623,14 +627,17 @@ def run(self):
         remote = self.getProperty('github.head.repo.full_name', 'origin').split('/')[0]
         project = self.getProperty('github.head.repo.full_name', self.getProperty('project'))
         branch = self.getProperty('github.head.ref', 'main')
+        base = self.getProperty('github.base.sha')
```

Comment:
this would be old base commit, at the time PR is created, right? Can you please explain little bit about where are we ensuring that we rebase on tip of tree (e.g.: main or destination branch), is it in CheckOutSource?

Reply:
Right.

It's the `git rebase --ont`o command that rebases onto whatever our rebase_target is. That target will either be the revision the build that triggered us specified (ie whatever `ews_revision` is) or the tip of our parent branch as specified in `CheckoutSource` via `got_revision`. The reason we are extracting the base here is because we need to tell git which set of commits to rebase, that is, we want all commits from whatever the base commit was at the time or PR creation to the `HEAD` commit of the PR. Basically:
`git rebase --onto <rebase_target> <base> <branch>` means "Take all commit from to (excluding itself) and re-do them with <rebase_target> as the root instead"